src/main/webapp/js/wikiSearch.js

```diff
@@ -0,0 +1,38 @@
+function searchOnWiki(){
+  fetch('/logo-detect').then(function(response) {
+    return response.json();
```

Comment:
Over here you assume that the JSON in the response will correspond to the request that you've made before. There is no way to guarantee that. If you test it on your local machine, where you have only one serving machine and only one user (you) it *will* work. However, if you run it on AppEngine, where the servlet will run on multiple machines the `logoJsonArray` will be private data, different on each serving machine.

It's quite possible that your request to upload the image and the response to /logo-detect will be served by different machines (or processes) and each of those will have different value for that parameter.